agent/structs/check_definition.go

```diff
@@ -42,6 +43,96 @@ type CheckDefinition struct {
 	OutputMaxSize                  int
 }
 
+func (t *CheckDefinition) UnmarshalJSON(data []byte) (err error) {
+	type Alias CheckDefinition
+	aux := &struct {
+		// Parse special values
+		Interval                       interface{}
+		Timeout                        interface{}
+		TTL                            interface{}
+		DeregisterCriticalServiceAfter interface{}
+
+		// Translate fields
+		ArgsCamel                           []string    `json:"args"`
```

Comment:
Is this what one of the FixupCheck functions somewhere else used to do? If so this could probably be named differently to indicate this isn't a snake -> camel case fix but something else. Maybe a comment would help too.